File: BinaryTreeInorderTraversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        List<int> result = new List<int>();
        
        traversal(root, result);
        
        return result;
    }
    
    public List<int> traversal(TreeNode node, List<int> list) {
        if(node == null)
            return list;
        
        if(node.left != null)
            traversal(node.left, list);
        
        list.Add(node.val);
        
        if(node.right != null)
            traversal(node.right, list);
        
        return list;
    }
}

File: BinaryTreeInorderTraversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    let bst = new BST(root);
    
    
    return bst.InOrderTraversal();
};

class BST {
    constructor(root = null){
        this.root = root;
    }

    InOrderTraversal = () => {
        let result = [];

        if(this.root)
            this.inOrderTraversal(this.root, result);

        return result;
    }

    inOrderTraversal = (node, result) => {
        if(!node)
            return;

        this.inOrderTraversal(node.left, result);

        result.push(node.val);

        this.inOrderTraversal(node.right, result);

        return result;
    }
}
